lib/ae_mdw/aex141.ex

```diff
@@ -0,0 +1,110 @@
+defmodule AeMdw.Aex141 do
+  @moduledoc """
+  Returns NFT info interacting with AEX-141 contracts or from transfer history.
+  """
+
+  alias AeMdw.AexnContracts
+  alias AeMdw.Collection
+  alias AeMdw.Db.Model
+  alias AeMdw.Db.State
+  alias AeMdw.Error
+  alias AeMdw.Error.Input, as: ErrInput
+  alias AeMdw.Util
+
+  import AeMdwWeb.Helpers.AexnHelper, only: [enc_ct: 1]
+
+  require Model
+
+  @typep pagination :: Collection.direction_limit()
+  @typep cursor() :: binary()
+
+  @typep pubkey :: AeMdw.Node.Db.pubkey()
+  @type token_id :: integer()
+
+  @ownership_table Model.NftOwnership
+
+  @spec fetch_nft_owner(pubkey(), token_id()) :: {:ok, pubkey()} | {:error, Error.t()}
+  def fetch_nft_owner(contract_pk, token_id) do
+    with :ok <- validate_aex141(contract_pk),
+         {:ok, {:address, account_pk}} <-
+           AexnContracts.call_contract(contract_pk, "owner", [token_id]) do
+      {:ok, account_pk}
+    else
+      {:error, exception} ->
+        {:error, exception}
+
+      _invalid_call_return ->
+        {:error, ErrInput.ContractReturn.exception(value: enc_ct(contract_pk))}
+    end
+  end
+
+  @spec fetch_owned_nfts(State.t(), pubkey(), cursor(), pagination()) ::
+          {:ok, {cursor() | nil, [token_id()], cursor() | nil}} | {:error, Error.t()}
+  def fetch_owned_nfts(state, account_pk, cursor, pagination) do
+    case deserialize_cursor(cursor) do
+      {:ok, cursor_key} ->
+        {prev_cursor_key, nfts, next_cursor_key_tuple} =
```

Comment:
Why is one called `_key` and the other one `_tuple`? I'd rather describe values by what they represent (key) instead of what they are (tuple)

Reply:
the tuple one is the cursor used by `PaginatedPlug` with the `is_reversed?` check